cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp

```diff
@@ -195,9 +195,9 @@ Ice::SSL::SecureTransport::TransceiverI::initialize(IceInternal::Buffer& readBuf
                 }
                 else if (_trust)
                 {
-                    for (CFIndex i = 0, count = SecTrustGetCertificateCount(_trust.get()); i < count; ++i)
+                    if (SecTrustGetCertificateCount(_trust.get()) > 0)
```

Comment:
Unrelated fix?

Reply:
Yes, this was bogus we want the leaf certificate with index 0. The previous code overrides `_peerCertificate` for each cert in the chain which doesn't make sense.